powerscale/provider/access_zone_resource.go

```diff
@@ -325,7 +325,7 @@ func (r *AccessZoneResource) Update(ctx context.Context, req resource.UpdateRequ
 	if state.Name != plan.Name {
 		editValues.Name = plan.Name.ValueStringPointer()
 	}
-	if len(state.CustomAuthProviders.Elements()) != len(plan.CustomAuthProviders.Elements()) {
+	if !state.CustomAuthProviders.Equal(plan.CustomAuthProviders) {
```

Comment:
It seems `Equal` would return false if elements are same but orders are different, thus AuthProviders will be updated.
Could you please check the behaviors of `Equal` and if this aligns with PR description (reorder ...)?

Reply:
Yes. According to the issue description, 
```
Running terraform apply reports the changes are made, but the cluster shows they have not been and terraform plan still shows the desired changes.
```
user switch the AuthProvider order and it doesn't actually make change on PowerScale. I think he wants it to be updated even if the elements are same but order are different.
Use `Equal` so that when reorder the elements, it would give false and then update the AuthProvider field.

Reply:
Ok, updating resource helps to resolve state divergence.